### DISCUSSION TITLE
Question to `org.freedesktop.portal.Request::Response`.

### DISCUSSION THREAD

**Discussion by TornaxO7** [flatpak/xdg-desktop-portal#1326] [Questions] [answered]
Hello! I'm trying to get a screenshot through the [`org.freedesktop.portal.Screenshot.Screenshot`](https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.Screenshot.html#org-freedesktop-portal-screenshot-screenshot) function which returns me a handle. According to the docs, I should use this path with [`org.freedesktop.portal.Request::Response`](https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.Request.html#org-freedesktop-portal-request-response) but how I a supposed to get retrieve the actual path afterwards?

I'm using rust with [zbus](https://crates.io/crates/zbus) to implement all this and here is my code:
```rust
{
    let conn = Connection::session()?;

    let screenshot = ScreenshotProxyBlocking::new(&conn).unwrap();
    let options = HashMap::from([
        ("interactive", Value::Bool(false)),
        ("handle_token", Value::Str(Str::from(crate_name!()))),
    ]);
    let path = screenshot.screenshot("", options)?;

    let dest = Signature::try_from("as").unwrap();
    // since `org.freedesktop.portal.Request` is a signal
    conn.emit_signal(
        Some("org.freedesktop.portal.Desktop"),
        path,
        "org.freedesktop.portal.Request",
        "Response",
        &dest,
    )
    .unwrap();

    println!("here: '{}'", dest);

    panic!("Did it work?");
}
```
I'm a bit out of clue here what I'm supposed to do to retrieve the actual path to the screenshot. Any help would be appreciated! :)

**Comment by hfiguiere** [comment, marked as answer]
Save yourself time and use https://crates.io/crates/ashpd

Or at best, check its code :-)

**Reply by TornaxO7** [reply]
![peepoWoah](https://github.com/flatpak/xdg-desktop-portal/assets/50843046/fff9ff1a-9a64-4564-828c-bec9b8c9eae6)

Dayum. Holy moly, thank you for hinting this! Basically wasted 2 or 3 days and I could've fixed this within 2 Minutes, lol xD
